zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/identity/GroupAddEntityProcessor.java

```diff
@@ -85,12 +74,24 @@ final var record = command.getValue();
       return;
     }
 
+    final var groupId = record.getGroupId();
+    final var groupKey = record.getGroupKey();
+    final var persistedRecord = groupState.get(groupId);
+    if (persistedRecord.isEmpty()) {
+      final var errorMessage =
+          "Expected to update group with ID '%s', but a group with this ID does not exist."
+              .formatted(groupId);
+      rejectionWriter.appendRejection(command, RejectionType.NOT_FOUND, errorMessage);
+      responseWriter.writeRejectionOnCommand(command, RejectionType.NOT_FOUND, errorMessage);
+      return;
+    }
+
     final var entityKey = record.getEntityKey();
     final var entityType = record.getEntityType();
     if (!isEntityPresent(entityKey, entityType)) {
       final var errorMessage =
-          "Expected to add an entity with key '%s' and type '%s' to group with key '%d', but the entity does not exist."
-              .formatted(entityKey, entityType, groupKey);
+          "Expected to add an entity with key '%s' and type '%s' to group with ID '%s', but the entity does not exist."
+              .formatted(entityKey, entityType, groupId);
```

Comment:
🔧 We're going to have to change entity keys to ids as well. I don't see an obvious subtask for this atm but you'll run into it soon enough with the api tasks 😄 You can do it as part of a different task just wanted to highlight.

Reply:
I'll create a separate issue